### PR TITLE
feat(navigationbar): add NavigationBar plugin

### PR DESCRIPTION
**Pull request by John-Luke**

### CODE REVIEW

File: src/plugins/navigationbar.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+})
+export class NavigationBar {
+
+  /**
+   * Hide the navigation bar.
+   * @param autohide {boolean} 
+   * @param callbacks {Object} Optional callbacks
+   */
+  @Cordova()
+  static setUp(autohide?: boolean = false, callbacks?: {success?: Function, failure?: Function}): void { }
```
Comment (ihadeed) [inline src/plugins/navigationbar.ts]: This should return a Promise.

Also the decorator should include the following options:
```
@Cordova({
  callbackStyle: 'object',
  successName: 'success',
  errorName: 'failure'
})
```

**Comment by John-Luke** [comment]
Like this @ihadeed ?

File: src/plugins/navigationbar.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+   * Hide the navigation bar.
+   * @param autohide {boolean} 
+   * @param callbacks {Object} Optional callbacks
+   */
+  @Cordova({
+    callbackStyle: 'object',
+    successName: 'success',
+    errorName: 'failure'
+  })
+  static setUp(autohide?: boolean = false): void { }
```
Comment (ihadeed) [inline src/plugins/navigationbar.ts]: Should be:
`static setUp(autohide?: boolean = false): Promise<any> { return; }`

File: src/plugins/navigationbar.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  pluginRef: 'plugins.navigationbar',
+  repo: 'https://github.com/cranberrygame/cordova-plugin-navigationbar',
+  platforms: ['Android']
+})
+export class NavigationBar {
+
+  /**
+   * Hide the navigation bar.
+   * @param autohide {boolean} 
+   * @param callbacks {Object} Optional callbacks
```
Comment (ihadeed) [inline src/plugins/navigationbar.ts]: Remove this line and add: 
```
@return {Promise<any>}
```

**Comment by John-Luke** [comment]
@ihadeed

File: src/plugins/navigationbar.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+   * @param autohide {boolean}  
+   * @return {Promise<any>}
+   */
+  @Cordova({
+    callbackStyle: 'object',
+    successName: 'success',
+    errorName: 'failure'
+  })
+  static setUp(autohide?: boolean = false): Promise<any> { return; }
+
```
Comment (mebibou) [inline src/plugins/navigationbar.ts]: Aren't you missing `hideNavigationBar` definition?

Comment (ihadeed) [inline src/plugins/navigationbar.ts]: Yes you're right. This is missing: https://github.com/cranberrygame/cordova-plugin-navigationbar/blob/master/www/navigationbar.js#L12

File: src/plugins/navigationbar.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * import { NavigationBar } from 'ionic-native';
+ *
+ * let autoHide: boolean = true;
+ * NavigationBar.hide(autoHide);
+ * ```
+ */
+@Plugin({
+  name: 'NavigationBar',
+  plugin: 'cordova-plugin-navigationbar',
+  pluginRef: 'plugins.navigationbar',
```
Comment (ihadeed) [inline src/plugins/navigationbar.ts]: you're right.. it is `navigationbar`

**Pull request merged**